src/analysis/vector/geometry_checker/qgsgeometrycheckresolutionmethod.cpp

```diff
@@ -0,0 +1,44 @@
+/***************************************************************************
+    qgsgeometrycheckresolutionmethod.cpp
+     --------------------------------------
+    Date                 : January 2020
+    Copyright            : (C) 2020 Matthias Kuhn
+    Email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsgeometrycheckresolutionmethod.h"
+
+QgsGeometryCheckResolutionMethod::QgsGeometryCheckResolutionMethod( int id, const QString &name, const QString &description, bool isStable )
+{
+  mId = id;
+  mName = name;
+  mDescription = description;
+  mIsStable = isStable;
+}
+
+int QgsGeometryCheckResolutionMethod::id() const
+{
+  return mId;
```

Comment:
Better to move these to the header so that they can be potentially inlined?

Reply:
Is that what we do now?
I don't think this code is performance critical at all, so the only reason for me to do this would be "best practice".

Reply:
We mix and match everywhere. I think for something like this where it's:
1. trivial (i.e. just a direct getter/setter)
2. doesn't add any new includes to the header
3. isn't virtual
we should put them in the header.

Reply:
I would argue that it's an implementation detail and should be purely internal.
Qt even goes one step further and follows the pimpl principle and only ever adds things to headers where performance *is* an issue.
Adding things to headers adds to compiler time because it needs to be parsed in every file where it's included and increases the risk that the header changes which will possibly trigger a myriad of cpp files to be recompiled. And it brings us a step away from ABI stability (ok, that's a big wish, I admit).
Anyway, I'd prefer to keep headers as minimal as possible unless *really* required for good reasons like performance. My 2c.

Reply:
Neither have I, but since you started the discussion I thought let's finish it ;)